src/test/java/Pages/TablePage.java

```diff
@@ -0,0 +1,79 @@
+package Pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.Select;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TablePage extends BasePage {
+
+    private final String URL = "https://demo.seleniumeasy.com/table-sort-search-demo.html";
+
+    @FindBy(css = "select[name = 'example_length']")
+    WebElement selectEntriesCountElement;
+
+    @FindBy(xpath = "//table/tbody/tr")
+    List<WebElement> tableRowElements;
+
+    @FindBy(xpath = "//a[@class = 'paginate_button next']")
+    WebElement nextButton;
+
+    private final String tableElementFormat = "//table/tbody/tr[%s]/%s";
+
+    public TablePage(WebDriver driver) {
+        super(driver);
+    }
+
+    @Override
+    protected String GetURL() {
+        return URL;
+    }
+
+    public void selectEntriesCount(int count) {
+        Select selectElement = new Select(selectEntriesCountElement);
+        selectElement.selectByValue(String.valueOf(count));
+    }
+
+    public List<Employee> selectEntries(int ageMin, double salaryMax) {
+        List<Employee> employees = new ArrayList<>();
+
+        do {
+            employees.addAll(selectEmployees(tableRowElements, ageMin, salaryMax));
+
+            if (!nextButton.isDisplayed()) {
+                nextButton.click();
+            } else {
+                break;
+            }
+        }
+        while (true);
+
+        return employees;
+    }
+
+    private List<Employee> selectEmployees(List<WebElement> rows, int ageMin, double salaryMax) {
```

Comment:
Update code with help of the following information: findElement method can be executed on WebElement instance. In this case new element will be found inside current. F.e. (pseudo-code)
```
WebElement headerBlock = driver.findElement(By.xpath("//div[@class='header']"));
WebElement logoutLink = headerBlock.findElement(By.xpath(".//a")); //do not forget about dot for xpath in these cases
```
Tips: you will have for in for-each implementation